### PR TITLE
added calling the native gas estimator

### PR DESCRIPTION
**Pull request by sunce86**
Added native gas estimator.
Gasnow websocket gas estimator replaced with etherchain http version.

### CODE REVIEW

**Review by bh2smith — APPROVED** [review]

File: shared/src/gas_price_estimation.rs [hunk 1]
```diff
@@ -85,20 +83,16 @@ pub async fn create_priority_estimator(
...
-                    return Err(anyhow!("gas now estimator did not initialize in time"));
-                };
-                estimators.push(Box::new(estimator));
+                estimators.push(Box::new(GasNowGasStation::new(client.clone())))
             }
             GasEstimatorType::GnosisSafe => estimators.push(Box::new(
                 GnosisSafeGasStation::with_network_id(&network_id, client.clone())?,
             )),
             GasEstimatorType::Web3 => estimators.push(Box::new(web3.clone())),
+            &GasEstimatorType::Native => {
```
[the remark this answers is not in the document]
Comment (sunce86) [reply shared/src/gas_price_estimation.rs]: Fixed.

File: shared/src/gas_price_estimation.rs [hunk 2]
```diff
@@ -85,20 +83,16 @@ pub async fn create_priority_estimator(
...
+                ensure!(is_mainnet(&network_id), "Native only supports mainnet");
+                estimators.push(Box::new(web3.clone()))
```
[the remark this answers is not in the document]
Comment (sunce86) [reply shared/src/gas_price_estimation.rs]: Now there is. Managed somehow to not commit all the changes.

**Comment by vkgnosis** [comment]
LGTM. I would love to start using only the natvie estimator but before that I think we should run a test that compares its results to Blocknative for a day or so to make sure there aren't huge outliers or bugs in our implementation. Of course this PR just allows us to enable and it doesn't force it, so either way merging is fine.

**Review by vkgnosis — APPROVED** [review]

**Review by nlordell — APPROVED** [review]

**Comment by nlordell** [comment]
Very exciting!

**Pull request merged**